### ISSUE TITLE
Crash when using hoppers to put items into Quantum Storage

### ISSUE THREAD

**Issue by runfalk** [gigabit101/QuantumStorage#18]
When I placed a hopper on top of a storage unit filled with rotten flesh and put arrows in the hopper the game crashed and I have to manually remove the hopper with an editor. It crashes the server. I have replicated it in a creative world

I got the following crash dump:

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  dsurroundcore (DynamicSurroundings-1.8.9-2.0.4.6BETA.jar)
  ThaumcraftLoader (Thaumcraft-1.8.9-5.1.6.jar)
  LoadingPlugin (ChunkAnimator-MC1.8.8-1.1.jar)
  FMLPlugin (InventoryTweaks-1.60-beta-30.jar)
Contact their authors BEFORE contacting forge

// Shall we play a game?

Time: 2016-02-12 02:40
Description: Ticking block entity

java.lang.NullPointerException: Ticking block entity
    at net.minecraft.tileentity.TileEntityHopper.func_174919_a(TileEntityHopper.java:309)
    at net.minecraft.tileentity.TileEntityHopper.func_145883_k(TileEntityHopper.java:274)
    at net.minecraft.tileentity.TileEntityHopper.func_145887_i(TileEntityHopper.java:211)
    at net.minecraft.tileentity.TileEntityHopper.func_73660_a(TileEntityHopper.java:196)
    at net.minecraft.world.World.func_72939_s(World.java:1716)
    at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:544)
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:701)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:344)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:605)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:481)
    at java.lang.Thread.run(Thread.java:745)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
    at net.minecraft.tileentity.TileEntityHopper.func_174919_a(TileEntityHopper.java:309)
    at net.minecraft.tileentity.TileEntityHopper.func_145883_k(TileEntityHopper.java:274)
    at net.minecraft.tileentity.TileEntityHopper.func_145887_i(TileEntityHopper.java:211)
    at net.minecraft.tileentity.TileEntityHopper.func_73660_a(TileEntityHopper.java:196)

-- Block entity being ticked --
Details:
    Name: Hopper // net.minecraft.tileentity.TileEntityHopper
    Block type: ID #154 (tile.hopper // net.minecraft.block.BlockHopper)
    Block data value: 0 / 0x0 / 0b0000
    Block location: World: (42,51,-127), Chunk: (at 10,3,1 in 2,-8; contains blocks 32,0,-128 to 47,255,-113), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
    Actual block type: ID #154 (tile.hopper // net.minecraft.block.BlockHopper)
    Actual block data value: 0 / 0x0 / 0b0000
Stacktrace:
    at net.minecraft.world.World.func_72939_s(World.java:1716)
    at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:544)

-- Affected level --
Details:
    Level name: world
    All players: 1 total; [EntityPlayerMP['redacted'/1202360, l='world', x=40,60, y=50,00, z=-123,36]]
    Chunk stats: ServerChunkCache: 737 Drop: 0
    Level seed: 4662684718019201355
    Level generator: ID 00 - default, ver 1. Features enabled: true
    Level generator options: 
    Level spawn location: 52,00,64,00,252,00 - World: (52,64,252), Chunk: (at 4,4,12 in 3,15; contains blocks 48,0,240 to 63,255,255), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
    Level time: 24561835 game time, 3904825 day time
    Level dimension: 0
    Level storage version: 0x04ABD - Anvil
    Level weather: Rain time: 149208 (now: false), thunder time: 1369 (now: true)
    Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
Stacktrace:
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:701)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:344)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:605)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:481)
    at java.lang.Thread.run(Thread.java:745)

-- System Details --
Details:
    Minecraft Version: 1.8.9
    Operating System: Linux (amd64) version 3.19.0-32-generic
    Java Version: 1.8.0_72, Oracle Corporation
    Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 303516792 bytes (289 MB) / 1103626240 bytes (1052 MB) up to 2798125056 bytes (2668 MB)
    JVM Flags: 0 total; 
    IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 94
    FML: MCP 9.19 Powered by Forge 11.15.1.1738 22 mods loaded, 22 mods active
    States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
    UCHIJAAAA   mcp{9.19} [Minecraft Coder Pack] (minecraft.jar) 
    UCHIJAAAA   FML{8.0.99.99} [Forge Mod Loader] (forge-1.8.9-11.15.1.1738-universal.jar) 
    UCHIJAAAA   Forge{11.15.1.1738} [Minecraft Forge] (forge-1.8.9-11.15.1.1738-universal.jar) 
    UCHIJAAAA   bagginses{2.1.5b} [Bagginses] (Bagginses-1.8.9-2.1.5b.jar) 
    UCHIJAAAA   spawn_defuzzer{1.1.0} [Spawn Defuzzer] (defuzz-1.2.0.jar) 
    UCHIJAAAA   dsurround{2.0.4.6BETA} [Dynamic Surroundings] (DynamicSurroundings-1.8.9-2.0.4.6BETA.jar) 
    UCHIJAAAA   Harvest{1.8.9-1.0.0-2} [Harvest] (Harvest-1.8.9-1.0.0-2.jar) 
    UCHIJAAAA   inventorytweaks{1.60-beta-30-48a432c} [Inventory Tweaks] (InventoryTweaks-1.60-beta-30.jar) 
    UCHIJAAAA   IronChest{6.0.125.770} [Iron Chest] (ironchest-1.8.9-6.0.125.770.jar) 
    UCHIJAAAA   JEI{2.25.10.149} [Just Enough Items] (jei_1.8.9-2.25.10.149.jar) 
    UCHIJAAAA   journeymap{5.1.3} [JourneyMap] (journeymap-1.8.8-5.1.3-fairplay.jar) 
    UCHIJAAAA   LunatriusCore{1.1.2.32} [LunatriusCore] (LunatriusCore-1.8.9-1.1.2.32-universal.jar) 
    UCHIJAAAA   Morpheus{1.8.9-2.0.57} [Morpheus] (Morpheus-1.8.9-2.0.57.jar) 
    UCHIJAAAA   reborncore{1.4.1.15} [RebornCore] (RebornCore-1.8.9-1.4.1.15-universal.jar) 
    UCHIJAAAA   QuantumStorage{2.0.2} [QuantumStorage] (QuantumStorage-1.8.9-2.0.2.jar) 
    UCHIJAAAA   Roguelike{1.4.5} [Roguelike Dungeons] (RoguelikeDungeons-1.8.9-1.4.5.jar) 
    UCHIJAAAA   Schematica{1.7.7.143} [Schematica] (Schematica-1.8.9-1.7.7.143-universal.jar) 
    UCHIJAAAA   SimpleLabels{0.0.3} [SimpleLabels] (simplelabels-1.8.9-0.0.4.jar) 
    UCHIJAAAA   Baubles{1.1.3.0} [Baubles] (Baubles-1.8.9-1.1.3.0.jar) 
    UCHIJAAAA   Thaumcraft{5.1.6} [Thaumcraft] (Thaumcraft-1.8.9-5.1.6.jar) 
    UCHIJAAAA   Waila{1.6.0} [Waila] (Waila-1.6.0-B3_1.8.8.jar) 
    UCHIJAAAA   wawla{1.4.7} [What Are We Looking At] (Wawla-1.4.7-1.8+.jar) 
    Loaded coremods (and transformers): 
dsurroundcore (DynamicSurroundings-1.8.9-2.0.4.6BETA.jar)
  org.blockartistry.mod.DynSurround.asm.Transformer
ThaumcraftLoader (Thaumcraft-1.8.9-5.1.6.jar)

LoadingPlugin (ChunkAnimator-MC1.8.8-1.1.jar)
  lumien.chunkanimator.asm.ClassTransformer
FMLPlugin (InventoryTweaks-1.60-beta-30.jar)
  invtweaks.forge.asm.ContainerTransformer
    Profiler Position: N/A (disabled)
    Player Count: 1 / 20; [EntityPlayerMP['redacted'/1202360, l='world', x=40,60, y=50,00, z=-123,36]]
    Is Modded: Definitely; Server brand changed to 'fml,forge'
    Type: Dedicated Server (map_server.txt)
```

**Comment by gigabit101** [comment]
Crap I will fix tonight expect a new version by tomorrow

**Comment by RealGrep** [comment]
Yep, I just got the same issue, but was using Charset. I had it going from a double vanilla chest to the Charset shifter, to the Charset pipe attached to your storage unit. It was passing birch wood along. Same exact crash. Log available on request, but it's the exact same thing as runfalk's crash, so probably redundant.

I flipped B:removeErroringTileEntities and B:removeErroringEntities to true in the forge.cfg file so it would remove the entities instead of crashing the server. Otherwise, whenever I logged in, the server would crash.

Glad you're on it! Looking forward to a fix. Great mod, I mean, other than the wee server crash that is, but you know what I mean. Obviously I need to store over 2 billion birch wood. I mean, who doesn't! ;)
